### PR TITLE
feat: agent-js v2.0.0

### PR DESCRIPTION
**Pull request by peterpeterparker**
# Motivation

Bump agent-js v2.0.0 ([release notes](https://github.com/dfinity/agent-js/releases/tag/v2.0.0)) and update `HttpAgent` creation accordingly.

# Changes

- `npm run update:agent`
- Defer fetching root key to `HttpAgent.create` in the `createAgent` which per extension also improve the creation of the agent by also calling `syncTime`
- For `defaultAgent` simply replace contructor call with synchronous `HttpAgent.createSync`

### CODE REVIEW

**Comment by dskloetd** [comment]
Will this require changes in nns-dapp when we update the ic-js deps in nns-dapp?
Would it be a good idea to test this in nns-dapp before merging here in case it is more work to make it work in nns-dapp?

**Comment by peterpeterparker** [comment]
Let me put it in draft and I do some testing first.

I don't think it will require changes in NNS dapp.

**Comment by peterpeterparker** [comment]
Indeed does not require any particular changes in NNS dapp except adapt the agent similarly as in this PR in `dev.api.ts`.
See PR https://github.com/dfinity/nns-dapp/pull/5274

File: packages/utils/src/utils/agent.utils.ts [hunk 1]
```diff
@@ -34,16 +34,11 @@ export const createAgent = async ({
   verifyQuerySignatures?: boolean;
   retryTimes?: number;
 }): Promise<HttpAgent> => {
-  const agent: HttpAgent = new HttpAgent({
+  return await HttpAgent.create({
```
Comment (mstrasinskis) [inline packages/utils/src/utils/agent.utils.ts:37]: Out of curiosity, why not use `createSync` everywhere?

**Review by mstrasinskis — APPROVED** [review]
LGTM, thank you!

File: packages/utils/src/utils/agent.utils.ts (hunk 1 above)
Comment (mstrasinskis) [reply packages/utils/src/utils/agent.utils.ts:37]: Ah, thanks for the explanation.

**Pull request merged**